FlowCrypt/Functionality/Services/Account Server Services/EnterpriseServerApi.swift

```diff
@@ -48,52 +44,21 @@ class EnterpriseServerApi: NSObject, EnterpriseServerApiType {
         return decoder
     }()
 
-    private func constructUrlBase(emailDomain: String) -> String {
-        guard !Bundle.isDebugBundleWithArgument("--mock-fes-api") else {
-            return "http://127.0.0.1:8001/fes" // mock
-        }
-        return "https://fes.\(emailDomain)" // live
-    }
-
     private var messageUploadProgressHandler: ((Float) -> Void)?
 
-    func getActiveFesUrl(for email: String) async throws -> String? {
-        do {
-            guard let userDomain = email.emailParts?.domain,
-                  !EnterpriseServerApi.publicEmailProviderDomains.contains(userDomain) else {
-                return nil
-            }
-            let urlBase = constructUrlBase(emailDomain: userDomain)
-            let request = ApiCall.Request(
-                apiName: Constants.apiName,
-                url: "\(urlBase)/api/",
-                timeout: Constants.getActiveFesTimeout,
-                tolerateStatus: [404] // 404 tells the app that FES is disabled
-            )
-            let response = try await ApiCall.call(request)
+    private let email: String
+    let fesUrl: String?
 
-            if response.status == 404 {
-                return nil // FES is explicitly disabled
-            }
+    init(email: String) async throws {
+        self.email = email
 
-            guard isExpectedFesServiceResponse(responseData: response.data) else {
-                if Bundle.isEnterprise() { // on enterprise build, FES is expected to be running
-                    throw AppErr.general("Unpexpected response from FlowCrypt Enterprise Server")
-                }
-                return nil // on consumer installations, we only use FES if it returns as expected
-            }
+        let helper = EnterpriseServerApiHelper()
+        self.fesUrl = try await helper.getActiveFesUrl(for: email)
```

Comment:
I wonder - could there be situations when `EnterpriseServerApi` is initiated but a call to check if FES is active is not actually needed in that moment? We could maybe instead have `fesUrl` as a lazy computed property?